### ISSUE TITLE
Canvas pannable and draggable

### ISSUE THREAD

**Issue by Lindsay89** [beautifulinteractions/beautiful-react-diagrams#78]
**Describe the bug**
Hi, we have published the 0.5 version of 'beautiful-react-diagram' library, but we recently noticed that this version has a bug.
Today we will patching the library to the old 0.4 version, the 0.5 will be deprecated.
So, for now, please use the old version, till the new one will be released.
We are working on this new version, the 0.6, that will have a new component, called Canvas, that will have the following API.

```
import Diagram, { Canvas, createSchema, useSchema, useCanvasState, CanvasControls } from 'beautiful-react-diagrams';
// the diagram model
const initialSchema = createSchema({
  nodes: [
    { id: 'node-1', content: 'Hey Jude', coordinates: [312, 27], },
    { id: 'node-2', content: 'Don\'t', coordinates: [330, 90], },
    { id: 'node-3', content: 'be afraid', coordinates: [100, 320], },
    { id: 'node-4', content: 'let me down', coordinates: [306, 332], },
    { id: 'node-5', content: 'make it bad', coordinates: [515, 330], },
  ],
  links: [
    { input: 'node-1',  output: 'node-2' },
    { input: 'node-2',  output: 'node-3' },
    { input: 'node-2',  output: 'node-4' },
    { input: 'node-2',  output: 'node-5' },
  ]
});
const DiagramExample = () => {
  const [canvasState, handlers] = useCanvasState(); // creates canvas state
  const [schema, { onChange }] = useSchema(initialSchema); // creates diagrams schema
  return (
    <div style={{ height: '30rem' }}>
      <Canvas {...canvasState} {...handlers}>
        <Diagram schema={schema} onChange={onChange} />
        <CanvasControls />
      </Canvas>
    </div>
  );
};
<DiagramExample />
```

And will looks like

<img width="957" alt="Screenshot 2020-11-25 at 11 15 44" src="https://user-images.githubusercontent.com/59820753/100224384-0e92f080-2f1d-11eb-98f8-ef0aadfba26c.png">

**Comment by alimirakim** [comment]
A pan-zoom was exactly what I needed for the project I'm using this with! So excited to hear it's already been being worked on, and look forward to the release :) . Thank you for the hard work. Thought I might have to figure out how to do it myself hahaha.

**Comment by antonioru** [comment]
@alimirakim 

Yes a way better pan and zoom functionality will be release in the next version, the `0.6.0`. 
Unfortunately we don't have a release date yet but it's on its way. 

If you need to have a look on how it look so far, be please pull the code from the `feature/canvas-ext` branch and feel free to contribute. 

About the hard work, I appreciate your comment, thank you... I must say, we worked on this as a company but some important  improvements come from the community and the help of splendid contributors like @9renpoto, @genbs and @joancc

**Comment by dfirence** [comment]
@Lindsay89 

Thanks to you and the team for this awesome resource, question.  When would ETA for Zoom be possibly released?

I am approaching a project start and would like to know if I can use this library

**Comment by RickyMarou** [comment]
Just started playing with the lib for an hour and i just want to say thank you for your work. The DX is 💯 and it's super easy to hit the ground running building Diagrams. 

Looking forward to the zoom/pan feature as I believe this is a key feature missing at the moment.
Thanks you for your work and I hope for a successful release 🙏

**Comment by dbflakes** [comment]
Thank you!
Really looking forward to the new zoom functionality.

**Comment by mikhail-fedosenko** [comment]
Could you also add handling of diagram node scrolling? E.g. diagram root element has non-zero `scrollLeft`, `scrollTop` because of overflow.
At the moment this causes visual glitches (because scrollLeft, scrollTop does not taken into account in `getEntityCoordinates.ts`.
<img width="524" alt="image" src="https://user-images.githubusercontent.com/1673511/189612311-37f53bbe-e020-4088-b7e2-8667f8d18a2a.png">
